### ISSUE TITLE
Jellyfin playback reporting stopped picking up songs played via sonixd

### ISSUE THREAD

**Issue by internetfreak** [jeffvli/sonixd#187]
Opening this as an issue here because I don't know which place is more likely the one I should be reporting to, but all things after another

**Describe the bug**
When listening to songs using sonixd up until v0.9 (or earlier, don't remember exactly) the playback was correctly noted via the jellyfin playback reporting plugin. With v0.10, no playback is being reported anymore, I assume that the changes made to scrobbling are responsible for this as the plugin might use some other mechanism to detect playbacks than sonixd now uses to scrobble to jellyfin.
Of course I have scrobbling activated, I can see my current song, progress etc on the dashboard. Screenshots attached below.

**To Reproduce**
1. Using an earlier version of Sonixd, activate scrobbling and install the jellyfin playback reporting plugin, if not present
2. Listen to any song
3. Notice, that the playback is being counted
4. Use Sonixd from v0.10 upwards and repeat steps 2 and 3. You notice that the last reported playback is the one you did earlier

**Expected behavior**
Playback of songs gets correctly scrobbled so that plugins can also pick it up (maybe it is already and the plugin is at fault, that's what I'm unsure about)

**Screenshots**
Screenshot of Sonixd playing a song
![image](https://user-images.githubusercontent.com/19251027/149616464-6aaebba0-e803-4677-adb4-69e3fea323b6.png)
Notice that it's reported correctly in the dashboard, so something is being scrobbled to jellyfin
![image](https://user-images.githubusercontent.com/19251027/149616471-d4727766-918e-41ce-9d3e-0087535f3c72.png)
Screenshot of how it looks for the playback reporting plugin. You see that my last played song was something I played via my iPhone and not the song I just started
![image](https://user-images.githubusercontent.com/19251027/149616474-4b5dc417-3484-4096-86d3-e7497acc7a58.png)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Version: 0.11.0

**Comment by jeffvli** [comment]
You're most likely correct that the [scrobbling changes](https://github.com/jeffvli/sonixd/issues/126) I made for 0.10.0 caused it to not work anymore.

Is [this](https://github.com/jellyfin/jellyfin-plugin-playbackreporting) the plugin that you're using? I'll need to double check, but my initial assumption is that it requires the `/sessions/playing` request to be sent for the song to be marked as played within the plugin. The reason it was removed was because it causing multiple plays to be recorded for the native Jellyfin playback statistics tracker, but I can check again to see if there's a feasible way to add it back.

**Comment by internetfreak** [comment]
Thanks for the fast reply. This is what I inferred from the respective bug reports, but I hesitated some time to open up that issue due to uncertainty where it belongs.

You are correct, this is the plugins I use for tracking. Even though I am using the server mostly for myself alone, I like tracking data about my usage, it's sometimes quite interesting to reflect what I did and I also used it to track which things I already listened to, when I mass-added a whole bunch of new stuff (as jellyfin sadly does not show what you listened to similar to movies/shows)

I don't know exactly how that request works but maybe it's enough to send that request at the time when the current scrobbling method marks the song as played? Or does this alone already cause the double counting within the native tracker?

**Comment by jeffvli** [comment]
Completely agree with you, I like to track my listening habits as well!

Before jumping to any conclusions, I'll need a bit of time to check the exact conditions in which the playback reporting plugin detects a song as "played".

This includes things like:
1. If it's checking for a certain percentage of the song played
2. What endpoint it's using to check it
3. If it's marking it as "played" immediately after starting playback (if it does this, it's most likely just using the `/sessions/playing` request since that fires off as soon as the song is started on the Web UI)

If you could do a quick test on the behavior of the plugin when you use the Web UI, that would be helpful for me to review before I install and try it out myself.

**Comment by internetfreak** [comment]
I did what you requested from me and from what I can see, your assumptions are right. I picked a random song, let it play via Web UI for about two seconds and captured the requests the UI made when I clicked start/stop. As soon as I started the song, it got already registered as play within the plugin with a duration of 31 seconds (the plugin can record durations as well to show which media (types) got played and how long).
The Firefox developer console lists a few web requests and aside from requesting the media file and the image for it, I can see that the `/sessions/playing` endpoint is triggered as well as the `/sessions/playing/progress` so I would expect the plugin to listen for events on those two endpoints for it's magic to work.
I think that should help you with your investigations. Should you require more data, I can provide more
